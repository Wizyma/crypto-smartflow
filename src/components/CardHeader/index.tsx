import { Row, Text } from '@zeit-ui/react';
import React from 'react';

export default function CardHeader({
  tab,
  onTabChange,
}: {
  tab?: string;
  onTabChange: (e: unknown) => void;
}) {
  console.log(tab);
  return (
    <Row justify="space-between" align="middle">
      <Text b>Your Wallet</Text>
      <div
        style={{
          display: 'flex',
        }}
      >
        <Text
          style={{ margin: '1rem 5px', cursor: 'pointer' }}
          b={tab === 'list'}
          id="list"
          onClick={onTabChange}
        >
          List
        </Text>
        <Text
          style={{ margin: '1rem 5px', cursor: 'pointer' }}
          b={tab === 'graph'}
          id="graph"
          onClick={onTabChange}
        >
          Graph
        </Text>
      </div>
    </Row>
  );
}
