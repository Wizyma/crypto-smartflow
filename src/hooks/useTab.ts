import { useCallback, useState } from 'react';

/**
 * @function useTab
 * @summary get current tab and make it easy to switch
 *
 * @export
 * @returns {tab, onTabChange}
 */
export default function useTab() {
  const [tab, setTab] = useState('list');
  const onTabChange = useCallback(
    (e) => {
      setTab(e.target.id);
    },
    [setTab],
  );

  return {
    tab,
    onTabChange,
  };
}
