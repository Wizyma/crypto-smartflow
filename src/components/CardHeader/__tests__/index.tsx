import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';

import useTab from 'hooks/useTab';
import CardHeader from 'components/CardHeader';

test(`render ${CardHeader.name} correctly`, async () => {
  const { result } = renderHook(() => useTab());
  const { asFragment, getByText } = render(
    <CardHeader tab={result.current.tab} onTabChange={result.current.onTabChange} />,
  );
  const fragment = asFragment();
  const node = getByText('List');
  const nodeGraph = getByText('Graph');

  expect(node.tagName).toBe('B');
  expect(nodeGraph.tagName).toBe('P');
  expect(fragment).toMatchSnapshot();
});
