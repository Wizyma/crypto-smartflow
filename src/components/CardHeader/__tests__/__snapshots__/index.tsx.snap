// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CardHeader correctly 1`] = `
<DocumentFragment>
  <div
    class="jsx-1311337904 row "
  >
    <b
      class="jsx-805937993  "
    >
      Your Wallet
    </b>
    <div
      style="display: flex;"
    >
      <b
        class="jsx-805937993  "
        id="list"
        style="margin: 1rem 5px; cursor: pointer;"
      >
        List
      </b>
      <p
        class="jsx-3390086683  "
        id="graph"
        style="margin: 1rem 5px; cursor: pointer;"
      >
        Graph
      </p>
    </div>
  </div>
</DocumentFragment>
`;
